### DISCUSSION TITLE
Analysis on full images not patches

### DISCUSSION THREAD

**Discussion by GioLetti** [MIC-DKFZ/nnUNet#2223] [Q&A]
Hi,

I'm working with nn-UNet to perform the segmentation of some MRI images. I've noticed (correct me if I'm wrong) that the 3d images get split in patches before being analyzed by the CNN, and for my specific task, I think this is important.

Is there the possibility of blocking such behavior? I would like the full 3d image to get passed to the UNet as it is.

Any help is appreciated,

Thanks.

Giorgio

**Comment by shub-garg** [comment]
You're correct that nnU-Net typically splits 3D images into patches before processing. If you want to process the full 3D image without splitting, you can adjust the patch size to match the full image dimensions. Here's how:

Set Patch Size: In your configuration, set the patch size to the full image dimensions.

Example:

```bash
patch_size = (full_image_depth, full_image_height, full_image_width)
batch_size = 1  # Handle one full image at a time
```

This will allow nnU-Net to process the full 3D image without splitting it into patches.
